core/modules/replica/SemanticMaps.h

```diff
@@ -394,6 +406,137 @@ using WorkerChunkDatabaseMap =
      }
  }
 
+/**
+ * One-directional comparison of dictionaries of: worker-database-chunk
+ *
+ * The method will also report keys which aren't found in second dictionary.
+ *
+ * NOTE: the output dictionary will be modified even if the method
+ *       will not find any differences.
+ *
+ * @param one       - input dictionary to be compared with the second one
+ * @param two       - input dictionary to be compared with the first one
+ * @param inOneOnly - output dictionary with elements of the first map which
+ *                    are not found in the second map
+ *
+ * @return 'true' if different
+ */
+ template<typename T>
+ bool diff(WorkerDatabaseChunkMap<T> const& one,
+           WorkerDatabaseChunkMap<T> const& two,
+           WorkerDatabaseChunkMap<T>& inOneOnly) {
+
+    inOneOnly.clear();
+    for (auto&& worker: one.workerNames()) {
+        if (not         two.workerExists(worker)) {
+            inOneOnly.insertWorker(worker,
+                                   one.worker(worker));
+            continue;
+        }
+        for (auto&& database: one.worker(worker).databaseNames()) {
+            if (not           two.worker(worker).databaseExists(database)) {
+                inOneOnly.atWorker(worker)
+                         .insertDatabase(database,
+                                         one.worker(worker).database(database));
+                continue;
+            }
+            for (auto&& chunk: one.worker(worker).database(database).chunkNumbers()) {
+                if (not        two.worker(worker).database(database).chunkExists(chunk)) {
+                    inOneOnly.atWorker(worker)
+                             .atDatabase(database)
+                             .insertChunk(chunk,
+                                          one.worker(worker).database(database).chunk(chunk));
+                }
+            }
+        }
+    }    
+    return not inOneOnly.empty();
```

Comment:
Hah, perfect! I sort of expected to see `inOneOnly.size() != 0` 😼

Reply:
👍